utilities/include/alps/numeric/tensors/tensor_base.hpp

```diff
@@ -316,10 +316,12 @@ namespace alps {
         /**
          * Inplace tensor multiplication
          */
-        template<typename S>
-        typename std::enable_if < std::is_same < S, tensorType >::value, tensorType & >::type operator*=(const S& rhs) {
+        template<typename S, typename Ct>
+        typename std::enable_if <
+          std::is_same < S, T >::value || std::is_same < T, std::complex < double>>::value
```

Comment:
why? what if T=complex<double> and S=complex<float>?
it is perfectly valid for *= operation, but it will not work in your case. You need to change this enable_if
in a way it properly handles this situation. For this you can use something like "std::is_convertible".

Could you also write a test that will take tensors with different scalar types and call this operation?
something like:
tensor<double, 2> a(N, N);
tensor<float, 2> b(N, N);
tensor<std::complex<double>, 2> c(N, N);
tensor<std::complex<float>, 2> d(N, N);

Initialize them with some values and then you should have something like this:

tensor<double, 2> ab = a * b;
tensor<std::complex<double>, 2> ac = a * c;
tensor<std::complex<float>, 2> bd = b * d;
tensor<std::complex<double>, 2> cd = c * d;

and check that the result of this product is correct.